### ISSUE TITLE
nil pointer evaluating interface {}.bookkeeper

### ISSUE THREAD

**Issue by johndlang** [apache/pulsar-helm-chart#337]
**Describe the bug**
When tying to install v3.0.0 I get a error when defining values for bookkeeper

**To Reproduce**
helm install pulsar-dev -f values/pulsar_values_dev.yaml apache/pulsar --version v3.0.0

with values.yaml: >

bookkeeper:
  configData:
    PULSAR_MEM: |
      -Xms512m -Xmx1g -XX:MaxDirectMemorySize=2g
  replicaCount: 3
  resources:
    requests:
      cpu: 1
      memory: 5Gi
  volumes:
    journal:
      name: journal
      size: 20Gi
      local_storage: false
      useMultiVolumes: true
      multiVolumes:
        - name: journal0
          size: 10Gi
          mountPath: /pulsar/data/bookkeeper/journal0
        - name: journal1
          size: 10Gi
          mountPath: /pulsar/data/bookkeeper/journal1
    ledgers:
      name: ledgers
      size: 100Gi
      local_storage: false
      useMultiVolumes: true
      multiVolumes:
        - name: ledgers0
          size: 50Gi
          mountPath: /pulsar/data/bookkeeper/ledgers0
        - name: ledgers1
          size: 50Gi
          mountPath: /pulsar/data/bookkeeper/ledgers1

Error: INSTALLATION FAILED: template: pulsar/templates/bookkeeper-statefulset.yaml:270:20: executing "pulsar/templates/bookkeeper-statefulset.yaml" at <.Values.bookkeeper.volumes.journal.selector>: nil pointer evaluating interface {}.bookkeeper

**Expected behavior**
Chart to install sucessfully 

**Screenshots**

**Desktop (please complete the following information):**
Helm v3.10.0 EKS 1.24

**Additional context**

**Comment by MarvinCai** [comment]
this is a bug, inside `{{ range }}`, the scope is already changed so .Values is no longer available, but seems it's already fixed in mainline, so can probably try that out.
```
{{ range .Values.bookkeeper.volumes.journal.multiVolumes }}
  - metadata:
      name: "{{ $fullname }}-{{ $bkComponent }}-{{ .name }}"
    spec:
      accessModes: [ "ReadWriteOnce" ]
      resources:
        requests:
          storage: {{ .size }}
    {{- if .storageClassName }}
      storageClassName: "{{ .storageClassName }}"
    {{- end }}
    {{- with .Values.bookkeeper.volumes.journal.selector }}
      selector:
        {{- toYaml . | nindent 8 }}
    {{- end }}
  {{- end }}
```

**Comment by michaeljmarshall** [comment]
Is this an issue with https://github.com/apache/pulsar-helm-chart/pull/286?

**Comment by johndlang** [comment]
@MarvinCai I don't think the fix has been merged ?

**Comment by lhotari** [comment]
Fixed by #342
